src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsysreflect/QSysReflectCommunicator.java

```diff
@@ -0,0 +1,745 @@
+				logger.debug("Populating system information data");
+			}
+			Map<String, String> deviceNameAndModelMap = new HashMap<>();
+			synchronized (aggregatedDeviceList) {
+				for (AggregatedDevice aggregatedDevice : aggregatedDeviceList) {
+					deviceNameAndModelMap.put(aggregatedDevice.getDeviceName(), aggregatedDevice.getDeviceModel());
+				}
+			}
+			synchronized (systemResponseList) {
+				for (SystemResponse systemResponse : systemResponseList) {
+					// Apply filter for system information that match the devices' name
+					if (deviceNameAndModelMap.containsKey(systemResponse.getCoreName())) {
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.SYSTEM_ID.getName()), String.valueOf(systemResponse.getId()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.SYSTEM_CODE.getName()), String.valueOf(systemResponse.getCode()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.SYSTEM_STATUS.getName()), String.valueOf(systemResponse.getStatusString()));
+						if (systemResponse.getNormalAlert() != null && systemResponse.getFaultAlert() != null && systemResponse.getUnknownAlert() != null && systemResponse.getWarningAlert() != null) {
+							stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.ALERTS_NORMAL.getName()), String.valueOf(systemResponse.getNormalAlert()));
+							stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.ALERTS_WARNING.getName()), String.valueOf(systemResponse.getWarningAlert()));
+							stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.ALERTS_FAULT.getName()), String.valueOf(systemResponse.getFaultAlert()));
+							stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.ALERTS_UNKNOWN.getName()), String.valueOf(systemResponse.getUnknownAlert()));
+						}
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.DESIGN_NAME.getName()), String.valueOf(systemResponse.getDesignName()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.DESIGN_PLATFORM.getName()), String.valueOf(systemResponse.getDesignPlatform()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.UPTIME.getName()), handleNormalizeUptime(systemResponse.getUptime()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.CORE_NAME.getName()), String.valueOf(systemResponse.getCoreName()));
+						stats.put(String.format("%s#%s", systemResponse.getName(), QSysReflectSystemMetric.MODEL.getName()), deviceNameAndModelMap.get(systemResponse.getCoreName()));
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Filter the list of aggregated devices based on filter option in Adapter Properties
+	 */
+	private void getFilteredAggregatedDeviceList() {
+		populateDeviceStatusMessage();
```

Comment:
does API provide any ways to filter it? It's good to have it made right off the bat during the data collection, but we still need to collect the full payload each time. 
If not - then we can leave it as it is